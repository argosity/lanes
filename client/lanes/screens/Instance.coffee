Lanes.renderScreenTo = (selector, options)->
    return new Lanes.Screens.Instance(selector, options);


class Lanes.Screens.Instance

    constructor: (selector, options)->
        this.viewport = new Lanes.Views.Viewport({ selector: selector, instance: this });
        Lanes.Data.Bootstrap.initialize(options);
        Lanes.$(document).ready => @boot(options)

    boot: (options)->
        this.root = Lanes.$( this.viewport.selector );
        this.root.data().workspace = this;
        this.viewport.root = this.root;

        Lanes.lib.ResizeSensor(this.root[0],  _.bind( _.debounce( =>
            @viewport.set({ width: this.root.width(), height: this.root.height() });
        , 250 ), this) );

        this.root.addClass('lanes root');
        this.root.tooltip({
            viewport: '.lanes'
            selector: '[data-tooltip-message]'
            title: => @getAttribute('data-tooltip-message')
        });
        Lanes.Views.Keys.initialize();
        Lanes.Data.PubSub.initialize() if options.pub_sub;
        
        view = Lanes.getPath(options.root_view);
        if view
            this.displayInitialView(view);
        else
            definition=Lanes.Data.Screens.all.findWhere({view: options.root_view});
            if definition
                definition.getScreen().then( (screen)=>
                    # break out of the promise so an errors during render get thrown properly
                    _.defer( => @displayInitialView(screen); )
                ,(msg)->
                    Lanes.fatal("Unable to load initial screen ${options.root_view}", msg);
                )
            else
                Lanes.fatal(options.root_view + " doesn't exist!");

    displayInitialView:(view)->
        this.view = new view({parent: this, model: this.viewport}).render();
        this.viewport.el = this.view.$el;

        this.root.append( this.view.el );
        Lanes.Extensions.fireOnAvailable(this);


