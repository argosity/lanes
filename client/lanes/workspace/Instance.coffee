Lanes.Workspace.create = (selector, options, extension_data)->
    return new Lanes.Workspace.Instance(selector, options, extension_data);


class Lanes.Workspace.Instance

    constructor: (selector, options)->
        this.viewport = new Lanes.Views.Viewport({ selector: selector, instance: this });
        Lanes.Data.Bootstrap.initialize({
            csrf: options.csrf_token,
            root: options.api_path,
            data: options
        });
        Lanes.$(document).ready => @boot(options)

    boot: (options)->
        this.root = Lanes.$( this.viewport.selector );
        this.root.data().workspace = this;
        this.viewport.root = this.root;

        Lanes.lib.ResizeSensor(this.root[0],  _.bind( _.debounce( =>
            @viewport.set({ width: this.root.width(), height: this.root.height() });
        , 250 ), this) );

        this.root.addClass('lanes root');
        this.root.tooltip({
            viewport: '.lanes'
            selector: '[data-tooltip-message]'
            title: => @getAttribute('data-tooltip-message')
        });
        Lanes.Views.Keys.initialize();
        Lanes.Data.PubSub.initialize() if options.pub_sub;
        
        view = Lanes.getPath(options.root_view);
        if view
            this.displayInitialView(view);
        else
            definition=Lanes.Data.Screens.all.findWhere({view: options.root_view});
            if definition
                definition.getScreen().then( (screen)=>
                    # break out of the promise so an errors during render get thrown properly
                    _.defer( => @displayInitialView(screen); )
                ,(msg)->
                    Lanes.fatal("Unable to load initial screen ${options.root_view}", msg);
                )
            else
                Lanes.fatal(options.root_view + " doesn't exist!");




    displayInitialView:(view)->
        this.view = new view({parent: this, model: this.viewport}).render();
        this.viewport.el = this.view.$el;

        this.root.append( this.view.el );
        Lanes.Extensions.fireOnAvailable(this);


