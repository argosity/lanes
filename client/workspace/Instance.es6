Lanes.Workspace.create = function(selector, options, extension_data){
    return new Lanes.Workspace.Instance(selector, options, extension_data);
};

Lanes.Workspace.Instance = class Instance {

    constructor(selector, options, extension_data){
        _.extend(this,options);

        this.viewport = new Lanes.Views.Viewport({ selector: selector, instance: this });
        Lanes.Data.Bootstrap.initialize({
            csrf: options.csrf_token,
            root: options.root_path,
            data: extension_data
        });
        Lanes.$(document).ready( ()=> {
            this.boot();
        });
    }



    boot(){
        this.root = Lanes.$( this.ui.selector );
        this.root.data().workspace = this;
        this.ui.root = this.root;

        Lanes.lib.ResizeSensor(this.root[0],  _.bind( _.debounce( function(a,b,c){
            this.ui.set({ viewport_width: this.root.width(), viewport_height: this.root.height() });
        }, 250 ), this) );

        this.root.addClass('lanes root');
        this.root.tooltip({
            viewport: '.lanes',
            selector: '[data-tooltip-message]',
            title: function(){ return this.getAttribute('data-tooltip-message'); }
        });

        var view = Lanes.getPath(this.view);
        if (!view){
            Lanes.fatal(this.view + " doesn't exist!");
        }
        this.view = new view({parent: this, model: this.ui}).render();
        this.ui.viewport = this.view.$el;

        this.root.append( this.view.el );

        Lanes.Views.Keys.initialize();
        Lanes.Data.PubSub.initialize();
        Lanes.Extensions.fireOnAvailable(this);

        this.login_dialog = new Lanes.Views.LoginDialog({ parent: this });

        // var screen = Lanes.Data.Screens.all.findWhere({ id: 'user-maintenance' });
        // screen.display();
    }
};
