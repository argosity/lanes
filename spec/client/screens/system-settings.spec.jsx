import React from 'react'; // eslint-disable-line no-unused-vars
import Settings from 'hippo/screens/system-settings';
import Instance from 'hippo/screens/instance';
import { getTestScreen } from '../test-models';


describe('SystemSettings Screen', () => {
    let definition;
    let instance;

    beforeEach(() => {
        definition = getTestScreen();
        instance = new Instance({ definition, isActive: true });
    });

    it('renders', () => {
        const settings = mount(<Settings screen={instance} />);
        expect(settings).toHaveRendered('Screen');
        settings.unmount();
    });
});
