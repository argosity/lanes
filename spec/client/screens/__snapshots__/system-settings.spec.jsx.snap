// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SystemSettings Screen renders 1`] = `
<div
  className="screen is-active"
  data-screen-id="test"
>
  <div
    className="grommetux-header__container grommetux-header__container--fixed grommetux-header__container--fill"
  >
    <div
      className="grommetux-header__mirror"
    />
    <div
      className="grommetux-header__wrapper"
    >
      <div
        className="grommetux-box grommetux-box--direction-column grommetux-box--responsive grommetux-box--pad-none grommetux-box--flex-off"
        id={undefined}
        onClick={undefined}
        role={undefined}
        style={Object {}}
        tabIndex={undefined}
      >
        <header
          className="grommetux-box grommetux-box--direction-row grommetux-box--align-center grommetux-box--pad-horizontal-none grommetux-box--pad-vertical-none grommetux-box--pad-between-small grommetux-header"
          id={undefined}
          onClick={undefined}
          role={undefined}
          style={Object {}}
          tabIndex={undefined}
        >
          <button
            aria-label={undefined}
            className="grommetux-button grommetux-button--primary"
            disabled={false}
            href={undefined}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
            type="button"
          >
            <span
              className="grommetux-button__icon"
            >
              <svg
                aria-label="save"
                className="grommetux-control-icon grommetux-control-icon-save grommetux-control-icon--responsive"
                height="24px"
                role="img"
                version="1.1"
                viewBox="0 0 24 24"
                width="24px"
              >
                <path
                  d="M3,2 L3,21 L21,21 L21,3 L12,3 L12,14 M8,11 L12,15 L16,11"
                  fill="none"
                  stroke="#000000"
                  strokeWidth="2"
                />
              </svg>
            </span>
            <span
              className="grommetux-button__label"
            >
              Save
            </span>
          </button>
        </header>
      </div>
    </div>
  </div>
  <h1
    className="grommetux-heading"
  >
    title
  </h1>
  <div
    className="row"
  >
    <div
      className="col-sm-4 col-xs-12"
    >
      <div
        className="asset form-field"
      >
        <div
          className="grommetux-form-field grommetux-form-field--size-medium"
          onClick={[Function]}
        >
          <label
            className="grommetux-form-field__label"
            htmlFor={undefined}
          >
            Logo
          </label>
          <span
            className="grommetux-form-field__contents"
          >
            <div
              className="drop-zone"
              onClick={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              style={Object {}}
            >
              <div>
                Drop a file here, or click to select one to upload.
              </div>
              <input
                accept={undefined}
                multiple={false}
                onChange={[Function]}
                style={
                  Object {
                    "display": "none",
                  }
                }
                type="file"
              />
            </div>
          </span>
        </div>
      </div>
    </div>
    <div
      className="col-sm-4 col-xs-12"
    >
      <div
        className="asset form-field"
      >
        <div
          className="grommetux-form-field grommetux-form-field--size-medium"
          onClick={[Function]}
        >
          <label
            className="grommetux-form-field__label"
            htmlFor={undefined}
          >
            Print Logo
          </label>
          <span
            className="grommetux-form-field__contents"
          >
            <div
              className="drop-zone"
              onClick={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              style={Object {}}
            >
              <div>
                Drop a file here, or click to select one to upload.
              </div>
              <input
                accept={undefined}
                multiple={false}
                onChange={[Function]}
                style={
                  Object {
                    "display": "none",
                  }
                }
                type="file"
              />
            </div>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
